### PR TITLE
feat: track docfx config in telemetry

### PR DESCRIPTION
**Pull request by herohua**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5441)

### CODE REVIEW

File: src/docfx/config/ConfigLoader.cs [hunk 1]
```diff
@@ -62,8 +62,10 @@ where glob(configPath)
             var envConfig = LoadEnvironmentVariables();
             var cliConfig = new JObject();
             JsonUtility.Merge(unionProperties, cliConfig, options?.StdinConfig, options?.ToJObject());
-            var docfxConfig = LoadConfig(errors, Path.GetFileName(configPath), File.ReadAllText(configPath));
-            var (xrefEndpoint, xrefQueryTags, opsConfig) = OpsConfigLoader.LoadDocfxConfig(docsetPath, _repository?.Branch ?? "master");
+            var docfxConfigFileName = Path.GetFileName(configPath);
+            var docfxConfigFileContent = File.ReadAllText(configPath);
+            var docfxConfig = LoadConfig(errors, docfxConfigFileName, docfxConfigFileContent);
+            var (docsetName, xrefEndpoint, xrefQueryTags, opsConfig) = OpsConfigLoader.LoadDocfxConfig(docsetPath, _repository?.Branch ?? "master");
```
Comment (yufeih) [inline src/docfx/config/ConfigLoader.cs]: `docsetName` is the same as `preloadConfig.Name`

File: src/docfx/config/ConfigLoader.cs [hunk 2]
```diff
@@ -86,6 +88,15 @@ where glob(configPath)
             var (configErrors, config) = JsonUtility.ToObject<Config>(configObject);
             errors.AddRange(configErrors);
 
+            // Track docfx config telemetry
+            Telemetry.TrackEvent(
+                docfxConfigFileName,
+                new Dictionary<string, string>
+                {
+                    ["Config"] = docfxConfigFileContent,
```
Comment (yufeih) [inline src/docfx/config/ConfigLoader.cs]: `docfxConfigFileContent` is YAML for `docfx.yml`, do you want to always produce json?
```suggestion
                    ["Config"] = docfxConfig.ToString(),
```

File: src/docfx/lib/log/Telemetry.cs [hunk 3]
```diff
@@ -69,6 +69,11 @@ public static void TrackBuildCommitCount(int count)
             s_buildItemCountMetric.TrackValue(count, TelemetryName.BuildCommits.ToString(), s_os, s_version, s_repo, s_branch);
         }
 
+        public static void TrackEvent(string name, Dictionary<string, string> properties, Dictionary<string, double> metrics = null)
```
Comment (yufeih) [inline src/docfx/lib/log/Telemetry.cs]: nit: consider a more descriptive method like `TrackConfig(config)`

Comment (herohua) [reply src/docfx/lib/log/Telemetry.cs]: so we prefer to hide telemetry specific logic in `Telemetry`? I think that's a good idea.

File: src/docfx/config/ConfigLoader.cs (hunk 2 above)
Comment (herohua) [reply src/docfx/config/ConfigLoader.cs]: I think it is okay to track different format for different input source - plus the event name is different if the config file is actually `docfx.yml`

**Pull request closed without merging**